sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -787,6 +790,17 @@ static void _handle_rtr_adv(gnrc_netif_t *netif, const ipv6_hdr_t *ipv6,
     if (dr == NULL) {
         return;
     }
+#if IS_ACTIVE(MODULE_DHCPV6_CLIENT)
+    if (rtr_adv->flags & NDP_RTR_ADV_FLAGS_M) {
+        if (IS_USED(MODULE_DHCPV6_CLIENT_IA_NA) &&
+            netif->ipv6.aac_mode & GNRC_NETIF_AAC_DHCP) {
+            dhcpv6_client_req_ia_na(netif->pid);
+        }
+        dhcpv6_client_set_conf_mode(DHCPV6_CLIENT_CONF_MODE_STATEFUL);
+    } else if (rtr_adv->flags & NDP_RTR_ADV_FLAGS_O) {
+        dhcpv6_client_set_conf_mode(DHCPV6_CLIENT_CONF_MODE_STATELESS);
+    }
```

Comment:
@miri64 Hmm, do we need an additional check here that the configuration mode isn't already stateful (cf. https://datatracker.ietf.org/doc/html/rfc4861#section-6.3.4)?

Reply:
Not sure where this is stated in the cited section (maybe provide a quote?), but https://datatracker.ietf.org/doc/html/rfc4861#section-4.2 says:

>                      If the M flag is set, the O flag is redundant and
>                      can be ignored because DHCPv6 will return all
>                      available configuration information.

So under that aspect, the way you check it here is already correct.

Reply:
Oh, sorry for being a bit imprecise – the case I was referring to was when you have multiple Routers sending contradicting RAs. If the client first receives an RA with an M flag and then one with an O flag, the configuration mode would be set to stateless although it could also operate in stateful mode. So I was wondering if the current should be checked first before switching to stateless mode.

Reply:
Mhh, that might indeed make sense.

Reply:
I tried to add a possible solution in d76b17379a149a53d6b8e902026efcee6667dffb.